# Homework Introduction to Python

$\textbf{Ubah nama di bawah yaa!}$

- Nama: Salsabila Mardhiyah

In [1]:
import numpy as np

# please import as much as you need

# Normal

## 1. Operasi Variabel

> Misal kita memiliki toko sepatu. Modal membeli 1 kodi sepatu adalah Rp 1.000.000. Setiap sepatu dijual seharga Rp 80.000. Buatlah variabel modal, kuantitas, dan harga_jual sesuai informasi di atas. Hitung profit yang didapatkan jika semua sepatu berhasil terjual, dan masukkan ke variabel profit.

Saran langkah-langkah pengerjaan
1. Convert terlebih dahulu kodi dalam jumlah barang (1 kodi = xyz buah sepatu)
2. Gunakan operasi matematika sederhana untuk menghitung profit menggunakan tiga buah variabel yang sudah ditentukan

In [63]:
# your code goes here!

modal = 1000000 #modal 1 kodi sepatu
kuantitas = 20 #1 kodi = 20 sepatu
harga_jual = 80000 #harga per sepatu
profit = harga_jual * kuantitas - modal
print('profit yang didapatkan:', profit)

profit yang didapatkan: 600000


Jadi, dari profit di atas, dapat disimpulkan bahwa profit yang didapatkan: 600000

## 2. Package Python dan Dictionary

> Kita akan coba mengakses data scientific article dari PLOS corpus
    Menggunakan API request dengan kode berikut : 
```
import requests
response = requests.get('http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')
```
**perhatikan tanda kutipnya tidak salah ketika meng-copy ke dalam python

Pertanyaannya : 
Convert response.content dari request tersebut dari JSON Format ke dalam Dictionary, dan print-kan key di dalamnya apa saja

Saran langkah-langkah pengerjaan
1. Ikuti code dari soal, kemudian ekstrak value dari variable response menggunakan dot (.) content menjadi sebuah variabel baru (dictionary)
2. Gunakan library json, convert hasil content menggunakan json.loads
3. Print keys dari dictionary tersebut


In [62]:
# your code goes here!

import requests
response = requests.get('http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')

# ekstrak value dari variable response menjadi sebuah variabel baru (dictionary)
dictionary = response.content

# convert hasil content menggunakan json.loads
import json
dictionary2 = json.loads(dictionary)

# Print keys dari dictionary tersebut
print(dictionary2.keys())

dict_keys(['response'])


Jadi, dari url link di atas, kami menyimpulkan bahwa memiliki keys ['response']

# Intermediate

## 3. Operasi dalam List

> Buatlah list dengan 10 nama kota di indonesia, kemudian:
1. Urutkan nama kota di dalam list berdasarkan abjad (a)
2. Ambil 4 kota pertama dari list tersebut (yang telah diurutkan) (b)
3. Ambil 3 kota terakhir dari list tersebut (yang telah diurutkan) (c)

Saran langkah-langkah pengerjaan
1. Tidak ada ketentuan nama kota (misal berdasarkan provinsi X atau pulau Y), dibebaskan untuk memilih. Buat nama-nama kota tersebut dalam sebuah list.
2. Gunakan fungsi sorted(X) untuk mengurutkan secara kecil-ke-besar (sesuai abjad), dimana X adalah contoh variabel list pada langkah (a)
3. Gunakan hasil poin (a), urutkan nama kota di dalam list berdasarkan abjad, untuk mengerjakan poin (b) dan (c). 

In [65]:
# your code goes here!

# Buatlah list dengan 10 nama kota di indonesia
visited = ['Sorong', 'Gorontalo', 'Makassar', 'Buleleng', 'Manggarai Barat', 'Langsa', 'Bintan', 'Banjarmasin', 'Batam', 'Padang Panjang']

# 1. Urutkan nama kota di dalam list berdasarkan abjad (a)
a = sorted(visited)
print('urutan abjad:', a)

# 2. Ambil 4 kota pertama dari list tersebut (yang telah diurutkan) (b)
b = a[:4]
print('4 kota pertama:', b)

# 3. Ambil 3 kota terakhir dari list tersebut (yang telah diurutkan) (c)
c = a[-3:]
print('3 kota terakhir:', c)

urutan abjad: ['Banjarmasin', 'Batam', 'Bintan', 'Buleleng', 'Gorontalo', 'Langsa', 'Makassar', 'Manggarai Barat', 'Padang Panjang', 'Sorong']
4 kota pertama: ['Banjarmasin', 'Batam', 'Bintan', 'Buleleng']
3 kota terakhir: ['Manggarai Barat', 'Padang Panjang', 'Sorong']


Jadi, dari operasi list di atas, dapat disimpulkan bahwa 
- urutan abjad: ['Banjarmasin', 'Batam', 'Bintan', 'Buleleng', 'Gorontalo', 'Langsa', 'Makassar', 'Manggarai Barat', 'Padang Panjang', 'Sorong']
- 4 kota pertama: ['Banjarmasin', 'Batam', 'Bintan', 'Buleleng']
- 3 kota terakhir: ['Manggarai Barat', 'Padang Panjang', 'Sorong']

## 4. Function dalam List

> Team sales memiliki list total transaksi dari 10 merchant pada bulan Juni 2020, yaitu: 
```
total_transaksi = [ 19, 35, 23, 25, 10, 15, 33, 45, 21, 30 ]
```
1. Trx minimum
2. Rata-rata 
3. Trx maksimal

Saran langkah-langkah pengerjaan
1. Copy variabel total_transaksi yang ada di soal ke notebook kita (atau ketik ulang boleh)
2. Kemudian, hitung menggunakan function operasi dalam list yang telah dipelajari di kelas pada topik list. Print hasilnya setiap poin (a), (b), dan (c)

In [70]:
# your code goes here!

total_transaksi = [ 19, 35, 23, 25, 10, 15, 33, 45, 21, 30 ]

# 1. Trx minimum
trx_minimum = min(total_transaksi)
print('transaksi minimum:', trx_minimum)

# 2. Rata-rata
rata_rata = sum(total_transaksi) / len(total_transaksi)
print('rata-rata transaksi:', rata_rata)

# 3. Trx maksimal
trx_maksimal = max(total_transaksi)
print('transaksi maksimal:', trx_maksimal)


transaksi minimum: 10
rata-rata transaksi: 25.6
transaksi maksimal: 45


Jadi, dari hasil operasi list di atas, dapat disimpulkan bahwa 
- nilai minimal: 10
- max: 45
- rata-rata: 25.6

# Advanced

## 5. NumPy Array

> Buatlah array dari 5 bilangan ganjil pertama (dimulai dari 1), Kemudian lakukan perintah berikut : 
1. Kali-kan array tersebut dengan 10
2. Bagi-kan array (hasil dari bagian a) dengan 3
3. Kali-kan array (hasil dari bagian b) dengan array baru :
`[-45, 43, 3, 56, 5]`
4. Carilah pada index ke-berapa nilai tertinggi dan terendah dari array (hasil dari bagian c) 

Saran langkah-langkah pengerjaan
1. HATI-HATI dalam pengerjaan, karena apabila poin sebelumnya salah, maka selanjutnya akan kebawa salah (misal (b) salah, maka poin (c) sampai (d) akan salah)
2. Untuk array 5 bilangan ganjil pertama boleh manual boleh menggunakan fungsi, dimulai dari 1
3. Ikuti perintah soal, gunakan numpy array, jangan list. Sesuaikan dengan permintaan soal pada setiap butir.



In [56]:
# your code goes here!

# Buatlah array dari 5 bilangan ganjil pertama (dimulai dari 1)
ganjil = np.arange(1, 10, 2)
print(ganjil)

# 1. Kali-kan array tersebut dengan 10
a = ganjil * 10
print(a)

# 2. Bagi-kan array (hasil dari bagian a) dengan 3
b = a / 3
print(b)

# 3. Kali-kan array (hasil dari bagian b) dengan array baru : [-45, 43, 3, 56, 5]
c = b * np.array([-45, 43, 3, 56, 5])
print(c)

# 4. Carilah pada index ke-berapa nilai tertinggi dan terendah dari array (hasil dari bagian c)
print('index nilai tertinggi: ', c.argmax())
print('index nilai terendah: ', c.argmin())

[1 3 5 7 9]
[10 30 50 70 90]
[ 3.33333333 10.         16.66666667 23.33333333 30.        ]
[-150.          430.           50.         1306.66666667  150.        ]
index nilai tertinggi:  3
index nilai terendah:  0


Jadi, dari hasil operasi numpy di atas, dapat disimpulkan bahwa 
- index nilai tertinggi:  3 
- index nilai terendah:  0

# Terimakasih, salam dari Bekasi 😀